paasta_tools/setup_tron_namespace.py

```diff
@@ -107,6 +108,22 @@ def main():
     failed = []
     skipped = []
 
+    master_config = tron_tools.create_complete_master_config(
+        cluster=args.cluster,
+        soa_dir=args.soa_dir,
+    )
+    if args.dry_run:
+        log.info(f"Would update {MASTER_NAMESPACE} to:")
+        log.info(f"{master_config}")
+        updated.append(MASTER_NAMESPACE)
+    else:
+        if client.update_namespace(MASTER_NAMESPACE, master_config):
+            updated.append(MASTER_NAMESPACE)
+            log.debug(f'Updated {MASTER_NAMESPACE}')
+        else:
+            skipped.append(MASTER_NAMESPACE)
+            log.debug(f'Skipped {MASTER_NAMESPACE}')
+
```

Comment:
Should still catch and report errors that happen here

Reply:
What if we didn't?

There is a kind of "race" between updating the master namespace (setting pools, etc), where it really has to happen first, before we try to update other stuff. 

What if we made updating the master configs a failing condition, and we "fail fast" here instead of moving on to update other stuff?

Reply:
Oh, it wasn't clear to me that `client.update_namespace` would raise an exception if the request was unsuccessful, but I see that it does. Failing fast sounds good to me.